### PR TITLE
Add [Steam] Service

### LINKED ISSUE #2121: Steam workshop item stats to shield
Any chance for getting Steam workshop item subscribers and other info on a shield? That info I believe can be freely retrieved so I assume it could be used for a shield

### PR DESCRIPTION
**Pull request by TagnumElite**
Originally #2130, however, some problems were made with git rebase which caused duplicated commits.

I believe I have addressed all the reported problems with the previous pull request. I do admit I could compress and make this code either more compact or easier to understand, I will do that if everybody needs it.

------------

This pull request adds 7 badges which come the data fetched from [Steams Web API](https://partner.steamgames.com/doc/webapi_overview) [ISteamRemoteStorage](https://partner.steamgames.com/doc/webapi/ISteamRemoteStorage) interface two end points [GetCollectionDetails](https://partner.steamgames.com/doc/webapi/ISteamRemoteStorage#GetCollectionDetails) and [GetPublishedFileDetails](https://partner.steamgames.com/doc/webapi/ISteamRemoteStorage#GetPublishedFileDetails)

These seven are:

1. Collection Size: Returns the number of files in a collection
2. File Size: Returns the file size of a workshop file
3. Release Date: Returns the initial release date (last update date not included)
4. Subscriptions: Returns the number of current subscribers
5. Favorites: Returns the current amount of favorites an file has received
6. Downloads: Returns the all-time amount of subscriptions a file has received
7. Views: Returns the view count of an file

And once again this pull request resolves #2121 

PS: @chris48s Hopefully this is it.

### CODE REVIEW

File: services/steam/steam.service.js [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+              .max(9)
+              .required(),
+          }).required()
+        )
+        .required(),
+    })
+    .required(),
+}).required()
+
+const collectionFoundOrNoteSchema = Joi.alternatives(
```
[the remark this answers is not in the document]
Comment (TagnumElite) [reply services/steam/steam.service.js]: I think I have a few other variables I wish to rename, but I will wait for another reviewer before making any more commits

**Comment by TagnumElite** [comment]
While I am at it, do you wish for me to split the file into more files corresponding to their category?

I have updated the var name and a few other ones.

It seems the Web API doesn't support ratings, will ask steam (and/or public forum) about this.

May I use your picture then because that is a much better, user-friendly, fewer steps approach to getting the URL!

**Comment by RedSparr0w** [comment]
>While I am at it, do you wish for me to split the file into more files corresponding to their category?

Sorry not quite sure what you mean by this?

>I have updated the var name and a few other ones.

👍 

>It seems the Web API doesn't support ratings, will ask steam (and/or public forum) about this.
I could scrape the website for the ratings but I don't think it would be worth it.
If you really want it I sure I can do it.

That's fine, we can always add it later once we find a better method to obtain them.

>May I use your picture then because that is a much better, user-friendly, fewer steps approach to getting the URL!
 
👍 No Problem

**Comment by TagnumElite** [comment]
> > While I am at it, do you wish for me to split the file into more files corresponding to their category?
> 
> Sorry not quite sure what you mean by this?
> 

I mean splitting different categories into their own folder i.e:

- `steam-other.service.js` = [`SteamCollectionFiles`, `SteamReleaseDate`, `SteamViews`]
- `steam-size.service.js` = [`SteamFileSize`]
- `steam-ratings.service.js` = [`SteamSubscriptions`, `SteamFavorites`]
- `steam-downloads.service.js` = [`SteamDownloads`]

At the same time renaming the following to be easier to understand:

1. `SteamCollectionFiles` => `SteamCollectionSize`
2. `SteamReleaseDate` => `SteamFileReleaseDate`
3. `SteamSubscriptions` => `SteamFileSubscriptions`
4. `SteamFavorites` => `SteamFileFavorites`
5. `SteamDownloads` => `SteamFileDownloads`
6. `SteamViews` => `SteamFileViews`

**Comment by RedSparr0w** [comment]
>I mean splitting different categories into their own folder i.e:

Thanks, I see what you mean now, yeah it may be worth splitting them into separate files,
as the steam badges definitely have a lot of room to grow.

Maybe something similar to these?:
- `steam-workshop-size.service.js`
- `steam-workshop-downloads.service.js`
_or_
- `steam-file-size.service.js`
- `steam-file-downloads.service.js`

just so it has a little more context as to what part of steam the badge is referring to, in case we add an app/game badge which shows similar info _(`steam-app-downloads.service.js`)_.

Or we could just put the workshop/file badges together in 1 seperate file `steam-workshop.service.js` or `steam-files.service.js`?

>At the same time renaming the following to be easier to understand:

Sounds good 👍

**Pull request merged**